erpnext/manufacturing/doctype/bom/bom.py

```diff
@@ -189,8 +189,8 @@ def validate(self):
 		self.validate_transfer_against()
 		self.set_routing_operations()
 		self.validate_operations()
-		self.update_exploded_items(save=False)
 		self.calculate_cost()
+		self.update_exploded_items(save=False)
```

Comment:
`calculate_cost` requires updated exploded items, which is why the old sequence was as such.
Can you explain the approach here ?

Reply:
Okay I assume its getting stale Items table row rate right ? then makes sense

Reply:
Yeah, Can we make this rate field read-only? bcoz rate is set by the system based on the "Rate Of Materials Based On" field and It also solves this wrong rate problem.

Reply:
@s-aga-r  So earlier any custom rate entered by the user was respected and rates were not reset as per the **Rate Of Materials Based On** field (valuation/last purchase rate, etc)
A recent fix changed that so it does respect the **Rate Of Materials Based On** field, but should users be allowed to tweak the rate?
My opinion aligns with yours, make the rate field read only. @rohitwaghchaure will that work or we missing a use case?

Reply:
@rohitwaghchaure @marination 

![image](https://user-images.githubusercontent.com/63660334/179348769-e8a16c57-7e0c-46c4-8ca4-dae98f2eadb3.png)